metadata/gc.go

```diff
@@ -257,15 +278,16 @@ func scanRoots(ctx context.Context, tx *bolt.Tx, nc chan<- gc.Node) error {
 }
 
 func references(ctx context.Context, tx *bolt.Tx, node gc.Node, fn func(gc.Node)) error {
-	if node.Type == ResourceContent {
+	switch node.Type {
+	case ResourceContent:
 		bkt := getBucket(tx, bucketKeyVersion, []byte(node.Namespace), bucketKeyObjectContent, bucketKeyObjectBlob, []byte(node.Key))
 		if bkt == nil {
 			// Node may be created from dead edge
 			return nil
 		}
 
 		return sendLabelRefs(node.Namespace, bkt, fn)
```

Comment:
do we need to skip the snapshot or content refs on the content?

Reply:
They are skipped by the flat type not being handled here.